petab/v2/lint.py

```diff
@@ -616,20 +766,24 @@ def append_overrides(overrides):
     # Add condition table parametric overrides unless already defined in the
     #  model
     parameter_ids.update(
-        p
-        for p in get_parametric_overrides(problem.condition_df)
-        if not problem.model.has_entity_with_id(p)
+        str(p)
+        for p in v2.conditions.get_condition_table_free_symbols(problem)
+        if not problem.model.has_entity_with_id(str(p))
     )
 
     # remove parameters that occur in the condition table and are overridden
     #  for ALL conditions
-    for p in problem.condition_df.columns[
-        ~problem.condition_df.isnull().any()
-    ]:
-        try:
-            parameter_ids.remove(p)
-        except KeyError:
-            pass
+    if problem.condition_df is not None:
+        ...
```

Comment:
I am not 100% sure if the old check ever made sense. If a parameter is overridden for some condition, it is no longer allowed to be in the parameters table. Not sure what the original intention was.

Reply:
The original code seems to do that; it removes parameters that are overridden for some condition?

Reply:
From the list of potentially required parameters for the parameters table, it removes those that are overridden in *all* conditions (the ones that are overridden only in some conditions do remain). I don't think it matters whether it's always overridden or just sometimes. A parameter overridden in the condition table is never required in the parameters table nor allowed, isn't it?


https://github.com/PEtab-dev/libpetab-python/blob/4be03c84dcb4a88972129582cee124f7a9b8ceb3/petab/v2/lint.py#L626-L634

Reply:
I thought the logic was rather: if a parameter is in the column header, but is not always overridden (either by specific values or other estimated parameters), then it could be estimated such that the non-overridden (`isnull`) cells take the estimated value. e.g. in your example, if `c2` was empty for `someParameter`, then `someParameter` could be estimated and that is the value for `c2`. For `c1`, `someParameter` is always overridden by `estimated_parameter_c1`.

Reply:
Oh, thanks, I see. That would be possible in principle, but I'd find that confusing.

PEtab specs are also clear that this would not be allowed:
https://github.com/PEtab-dev/PEtab/blob/b9e141dd75798d179c17262f085ed6cef8555b3e/doc/v1/documentation_data_format.rst?plain=1#L445-L449

Reply:
Great, I also thought it makes sense to just exclude all column header parameters :+1: Then the old code is a bug, thanks for removing!